java.base/src/main/java/jdk/internal/misc/Unsafe$_patch.java

```diff
@@ -48,4 +48,11 @@ public int pageSize() {
         }
         return UnsafeConstants.PAGE_SIZE;
     }
+
+    @Replace
+    private void checkNativeAddress(long address) {
+        // The original assumes a 4-byte address size
+        // and applies bitwise operators to validate.
+        // Will not work on platforms with word size < 4 bytes
+    }
```

Comment:
Actually, are you sure you still need this?  The code in Unsafe seems properly guarded, so maybe this is a symptom of another problem.

```
    private void checkNativeAddress(long address) {
        if (ADDRESS_SIZE == 4) {
            // Accept both zero and sign extended pointers. A valid
            // pointer will, after the +1 below, either have produced
            // the value 0x0 or 0x1. Masking off the low bit allows
            // for testing against 0.
            if ((((address >> 32) + 1) & ~1) != 0) {
                throw invalidInput();
            }
        }
    }
```

Reply:
uhm I tried to build with platform wasm-wasi and the bit shift was blowing up on `unboxLong` but you may be right that it may be a different root cause 🤔

Reply:
resolved by patching at the interpreter level. and. wait a minute, I am not even sure the `main` patch is necessary now 🤔